.github/workflows/build-toolchain.yml

```diff
@@ -317,6 +336,8 @@ jobs:
                 },
                 {
                   "arch": "arm64",
+                  "cpu": "aarch64",
```

Comment:
IIUC, the intent here is to build the arm64 toolchain from an arm64 host. So the arm64-built-on-x64 and x64-built-on-arm64 jobs should go away since these are building the same artifacts the arm64-built-on-arm64 and x64-built-on-x64 jobs. I believe this would take care of your issue with the naming conflict.

Reply:
@Steelskin We want to keep three builds: x64-built-on-x64, arm64-built-on-x64 (existing), arm64-built-on-arm64 (new) because the goal here is not just producing the release builds but keep those builds working. Initially we use the first two to produce the release binaries and the third just to ensure that it works.